backend/app/api/endpoints/knowledge_open.py

```
@@ -0,0 +1,530 @@
+
+@router.get(
+    "/list",
+    response_model=KnowledgeBaseListResponse,
+)
+@trace_sync("list_knowledge_bases_open", "knowledge.api")
+def list_knowledge_bases_open(
+    scope: str = Query(
+        default="all",
+        description="Scope of knowledge bases to return: 'personal' or 'all'",
+    ),
+    auth_context: AuthContext = Depends(get_auth_context),
```

Ruff (0.15.9)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)